### ISSUE TITLE
Cross platform Gitup

### ISSUE THREAD

**Issue by Thrilleratplay** [open-source-ideas/open-source-ideas#89]
[GitUp](https://github.com/git-up/GitUp) is an amazing git gui for OSX/MacOS and only OSX/MacOS.  A [Linux version has been requested](https://github.com/git-up/GitUp/issues/160) but would require a complete rewrite.  While there are many git guis out there, none have matched the ease GitUp allows visualizing multiple branches, cherry picking, squashing commits, editing commit messages swapping commits.

Create a version that can be used by developers on Linux, MacOS or Windows.

## Relevant Technology

* git (obviously)
* Probably Javascript/Electron/React native/Ionic/NativeScript/etc but the language and framework are not as important as keeping a single code base that can be used across different OSs.

## Who is this for

Any developer who uses git.

### Complexity

- `[ ]` Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- `[X]` Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- `[ ]` Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- `[ ]` Little work - A couple of days
- `[ ]` Medium work - A week or two
- `[X]` Much work - The project will take more than a couple of weeks and serious planning is required

---

Share on Twitter/other social media if you will. Something like

```
New project on Open Source Ideas Initiative: [fill in short description] – [fill in link]
```

**Comment by piedoom** [comment]
What about https://www.gitkraken.com/? If you're going the electron route this seems to be a very similar project.

**Comment by Thrilleratplay** [comment]
@piedoom gitkraken is not open source

**Comment by KOLANICH** [comment]
How about https://github.com/TortoiseGit/TortoiseGit ?

**Comment by FredrikAugust** [comment]
By the way, for a list of a lot of them, see the [official git webstie](https://git-scm.com/downloads/guis/).

**Comment by Thrilleratplay** [comment]
I have a used a few git GUIs, (git-gui, gitg, giggle, git-cola, git-tower, tortoiseGit)each has its strong points and weak points.  GitUp is unique.

**Comment by FredrikAugust** [comment]
Yeah, GitUp truly is something for itself, though I think Tower 2 can compare in a lot of ways. Too bad it's proprietary and only for mac/windows.

**Comment by KobbyMmo** [comment]
Project started already?

**Comment by FredrikAugust** [comment]
Don't think so :) 


Sent from my Samsung Galaxy smartphone.
null

**Comment by aksappy** [comment]
Has the work started yet? 
Can this be a plugin into VSCode?

**Comment by Thrilleratplay** [comment]
@aksappy Not that I am aware of.  As a plugin for VSCode or Atom sounds like a good idea.

**Comment by aksappy** [comment]
Never built one, but will have a stab at it..

**Comment by awmottaz** [comment]
How about [Sublime Merge](https://www.sublimemerge.com/)? Haven't tried it yet myself, but looks to be really well done.

**Comment by nickolay** [comment]
Sublime Merge [has UI for history rewriting](https://github.com/sublimehq/sublime_merge/issues/104), but it's far less pleasant to use than Gitup.

**Comment by Thrilleratplay** [comment]
Not to mention, as this is "open source ideas", a solution should be open source.

**Comment by minarch91** [comment]
меня потому что я мечтаю создать монету для Справедливости.Разработка и кодирование децентрализованной,саморегулируемой криптовалюты.Экономика которой является надежным и беспроигрышным решением базирующемся на трансформации ,,до судебно распределённого  , 100% Морального ущерба,(моральных потерь-убытка)>>трансформация>> в Материальные доходы. Я мечтаю о платформе для каждого пользователя блокчейна и криптовалюте на основе ETH которая может представлять собой универсальное приложение для запуска различных(национальных) квалификаций токенов,базирующихся на специализированных(национальных) смарт-контрактах,облегчающих нагрузку на национальные системы уголовного и гражданского правосудия ,толкающих(делающих выгодным) двигаться в сторону Справедливости на основе ,до судебной, гражданской пере адресации-распределении 100%-ов Морального ущерба, в смарт-контракте.Экономика Токена-справедливости становится привязанной к > общественной организации > моральному ущербу >Справедливости> банку. Я очень хотел бы чтобы Вы помогли мне с построением и оптимизацией алгоритма сети а также обширной практики настройки параметров алгоритма,чтобы было правильно. Всё это требует глубокого,детального понимания кода,который я надеюсь построить под Вашим руководством. Я прошу учиться у Вас и хочу достичь, в партнерстве с Вами ,взаимопонимания. minarch91.15.04.2019, 15:02, "aksappy" <notifications@github.com>:Never built one, but will have a stab at it..

**Comment by codymikol** [comment]
Are people still interested in this? It's one of the few things keeping me from going full on linux. I'll put a repo out there once I. finish up my MVP of the commit tab.

Progress so far, written in jetpack-compose.

![image](https://user-images.githubusercontent.com/13606342/138577028-d24999aa-3614-4011-a229-c4fe1b3a3474.png)

**Comment by codymikol** [comment]
Here is the repo if anyone is interested

 https://github.com/codymikol/git-down

**Comment by Thrilleratplay** [comment]
@codymikol I am eager to try this out.  First, perfect name.  Second, when you have a chance, can you add a License and README to the repo?  I noticed there were issues open but a README.md giving basic build instructions, current limitations and if needed, a warning saying "this is still in active development, not responsible for data loss, don't use without backup of repo".

**Comment by codymikol** [comment]
@Thrilleratplay sure, I'll add that after work tonight. Its still very much in development so I'd suggest not using this against anything you care about until a more stable release 😛

**Comment by codymikol** [comment]
Taken care of now btw
